openc3/lib/openc3/script/api_shared.rb

```diff
@@ -726,6 +728,12 @@ def _openc3_script_wait_implementation(target_name, packet_name, item_name, valu
       end
 
       return false, value
+    rescue NameError => error
+      if error.message =~ /uninitialized constant OpenC3::ApiShared::(\w+)/
```

Comment:
Will this still work with your Cosmos compatibility layer?

Reply:
No but probably ok.  However make sure you preserver the stack trace when raising a new error.